src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ITargetFrameworkMonikerComparer.cs

```diff
@@ -25,5 +25,7 @@ public interface ITargetFrameworkMonikerComparer : IComparer<TargetFrameworkMoni
         /// <param name="result">A merged TFM. For example, passing <c>net5.0-windows</c> and <c>net6.0</c> will result in <c>net6.0-windows</c>.</param>
         /// <returns>Whether the merge was successful.</returns>
         bool TryMerge(TargetFrameworkMoniker tfm1, TargetFrameworkMoniker tfm2, [MaybeNullWhen(false)] out TargetFrameworkMoniker result);
+
+        bool TryParse(string input, [MaybeNullWhen(false)] out TargetFrameworkMoniker tfm);
```

Comment:
This seems like an odd method to have on a comparer interface. It doesn't really have anything to do with comparing and results in the comparer being needed in odd places. It would be great if it could just live on the TargetFrameworkMoniker type, but I know NuGet dependencies make that hard.

How much work would it be to write our own parser that doesn't have a NuGet dependency? It seems like it shouldn't be _that_ hard. If that doesn't seem like a good option, maybe create a dedicated TargetFrameworkMoniker factory interface for this?

Reply:
I agree - I'm thinking we can fold a lot of the stuff the comparer was doing into the TargetFrameworkMoniker itself. I'll leave that for a separate PR to address though

Reply:
Here's an issue to track this: https://github.com/dotnet/upgrade-assistant/issues/453